src/main/java/de/zalando/aruha/nakadi/repository/kafka/NakadiKafkaConsumer.java

```diff
@@ -52,25 +54,26 @@ public NakadiKafkaConsumer(final KafkaFactory factory, final String topic, final
         if (eventQueue.isEmpty()) {
             pollFromKafka();
         }
-
         return Optional.ofNullable(eventQueue.poll());
     }
 
     @Override
     public Map<String, String> fetchNextOffsets() {
-        return topicPartitions.stream().collect(Collectors.toMap(topicPartition ->
-                        Integer.toString(topicPartition.partition()),
-                    topicPartition -> Long.toString(kafkaConsumer.position(topicPartition))));
+        return topicPartitions
+                .stream()
+                .collect(Collectors.toMap(
+                        topicPartition -> Integer.toString(topicPartition.partition()),
+                        topicPartition -> Long.toString(kafkaConsumer.position(topicPartition))));
     }
 
     private void pollFromKafka() throws NakadiException {
         try {
             final ConsumerRecords<String, String> records = kafkaConsumer.poll(pollTimeout);
-            eventQueue = StreamSupport.stream(records.spliterator(), false).map(record ->
-                                              new ConsumedEvent(record.value(), record.topic(),
-                                                  Integer.toString(record.partition()),
-                                                  Long.toString(record.offset() + 1))).collect(Collectors.toCollection(
-                                              Lists::newLinkedList));
+            eventQueue = StreamSupport
+                    .stream(records.spliterator(), false)
+                    .map(record -> new ConsumedEvent(record.value(), record.topic(),
+                            Integer.toString(record.partition()), Long.toString(record.offset() + 1)))
```

Comment:
I don't see a way to get rid of that. Strings should become numbers on kafka level (and vise versa).
Do you think that it will make sense to write some helper class that does the convertion?